src/nvim/testdir/test_usercommands.vim

```diff
@@ -312,3 +340,201 @@ func Test_use_execute_in_completion()
   call assert_equal('"DoExec hi', @:)
   delcommand DoExec
 endfunc
+
+func Test_addr_all()
+  command! -addr=lines DoSomething let g:a1 = <line1> | let g:a2 = <line2>
+  %DoSomething
+  call assert_equal(1, g:a1)
+  call assert_equal(line('$'), g:a2)
+
+  command! -addr=arguments DoSomething let g:a1 = <line1> | let g:a2 = <line2>
+  args one two three
+  %DoSomething
+  call assert_equal(1, g:a1)
+  call assert_equal(3, g:a2)
```

Comment:
https://github.com/neovim/neovim/pull/11318/files#diff-4aa0a7f67d049bd357dce116b348d1a1R350-R354
Commenting these out makes the test pass.